stages/01_process.py

```diff
@@ -0,0 +1,119 @@
+import logging
+import pandas as pd
+from pathlib import Path
+import pyarrow.lib as pa
+import re
+import zipfile
+
+logging.basicConfig(format="%(asctime)s %(message)s")
+logger = logging.getLogger(__name__)
+logger.setLevel("INFO")
+
+
+def create_unzip_dir():
+    logger.info("creating unzip/ directory")
+    unzip_dir = Path("unzip")
+    unzip_dir.mkdir(exist_ok=True)
+    return unzip_dir
+
+
+ZIP_DIR_PATTERN = re.compile(".*((?=tab3)|(?=ptm)|(?=chemtab)|(?=-4.4.232))")
+
+
+def unzip_file(f):
+    file_path = Path(f)
+    match = re.match(ZIP_DIR_PATTERN, file_path.stem)
+    if match is not None:
+        directory_base = match[0]
+        base_path = Path("unzip")
+        if directory_base[-1] == ".":
+            base_path /= Path(directory_base[:-1])
+        else:
+            base_path /= Path(directory_base)
+        if not base_path.exists():
+            base_path.mkdir()
+            logger.info("unzipping file at %s", base_path)
+    with open(file_path, "rb") as zip:
+        try:
+            zip_root = zipfile.ZipFile(zip)
+            zip_root.extractall(base_path)
+        except zipfile.BadZipFile:
+            logger.debug("Problem unzipping file", exc_info=1)
+
+
+DTYPES = {k: pd.StringDtype(storage="pyarrow") for k in
```

Comment:
These can also be treated as strings (some of the are integer IDs, but treating them as strings is fine):

```
"Entrez Gene Interactor A"
"Entrez Gene Interactor B"
"Related BioGRID Gene ID"
"Related Entrez Gene ID"
"Related Organism ID"
"Synonyms Interactor A"
"Synonyms Interactor B"
"Systematic Name Interactor A"
"Systematic Name Interactor B"
```